presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -960,6 +961,34 @@ public void finishDelete(Session session, TableHandle tableHandle, Collection<Sl
         return ImmutableMap.copyOf(views);
     }
 
+    @Override
+    public Map<String, Object> getSchemaProperties(Session session, CatalogSchemaName schemaName)
+    {
+        if (schemaExists(session, schemaName)) {
```

Comment:
I don't like this optionality. When you ask for properties of not existing object this method should fail. Returning empty map suggests that object exists but has no properties.

Same to the method below.

Reply:
Hmm... I was following other examples such as getView and getTable. But these *are* different.

Reply:
This is more like `io.prestosql.metadata.MetadataManager#getTableProperties`

Reply:
Added an error check to both methods.